sim/rogue/items_sets_pve.go

```diff
@@ -213,3 +215,171 @@ var ItemSetNightSlayerBattlearmor = core.NewItemSet(core.ItemSet{
 		},
 	},
 })
+
+var ItemSetBloodfangThrill = core.NewItemSet(core.ItemSet{
+	Name: "Bloodfang Thrill",
+	Bonuses: map[int32]core.ApplyEffect{
+		// Your opening moves have a 100% chance to make your next ability cost no energy.
+		2: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+
+			var affectedSpells []*core.Spell
+
+			aura := rogue.RegisterAura(core.Aura{
+				Label:    "Clearcasting (S03 - Item - T2 - Rogue - Damage 2P Bonus)",
+				ActionID: core.ActionID{SpellID: 467735},
+				Duration: time.Second * 15,
+				OnInit: func(aura *core.Aura, sim *core.Simulation) {
+					affectedSpells = core.FilterSlice(
+						rogue.Spellbook,
+						func(spell *core.Spell) bool {
+							return spell != nil && spell.Cost != nil && spell.Cost.CostType() == core.CostTypeEnergy
+						},
+					)
+				},
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+					core.Each(affectedSpells, func(spell *core.Spell) { spell.Cost.Multiplier -= 100 })
+				},
+				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+					core.Each(affectedSpells, func(spell *core.Spell) { spell.Cost.Multiplier += 100 })
+				},
+				OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
+					if aura.RemainingDuration(sim) == aura.Duration || spell.DefaultCast.Cost == 0 {
+						return
+					}
+					aura.Deactivate(sim)
+				},
+			})
+
+			core.MakePermanent(rogue.RegisterAura(core.Aura{
+				Label:    "S03 - Item - T2 - Rogue - Damage 2P Bonus",
+				OnSpellHitDealt: func(_ *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if !result.Landed() || !spell.ProcMask.Matches(core.ProcMaskMeleeOrRangedSpecial) {
+						return
+					}
+					if spell.SpellCode == SpellCode_RogueAmbush || spell.SpellCode == SpellCode_RogueGarrote {
+						aura.Activate(sim)
+					}
+				},
+			}))
+		},
+		// Increases damage dealt by your main hand weapon from combo-generating abilities by 20%
+		4: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			rogue.OnSpellRegistered(func(spell *core.Spell) {
+				//Currently not working as intended.  The below are tested to have worked.  MG and PK are subspells of Sinister Strike even though they are offhand attacks. SSL DoT confirmed to get the bonus as well.
+				if spell.SpellCode == SpellCode_RogueAmbush || spell.SpellCode == SpellCode_RogueBackstab || spell.SpellCode == SpellCode_RogueGhostlyStrike || spell.SpellCode == SpellCode_RogueHemorrhage || spell.SpellCode == SpellCode_RogueMainGauche || spell.SpellCode == SpellCode_RoguePoisonedKnife || spell.SpellCode == SpellCode_RogueSaberSlash || spell.SpellCode == SpellCode_RogueSaberSlashDoT || spell.SpellCode == SpellCode_RogueShadowStrike || spell.SpellCode == SpellCode_RogueSinisterStrike || (spell.SpellCode == SpellCode_RogueMutilate && spell.ActionID.Tag == 1) {
```

Comment:
Can we code this based on how we suspect/have gotten confirmation for how it should work?